cmd/main.go

```diff
@@ -1,11 +1,51 @@
 package main
 
 import (
+	"flag"
 	"log"
 	"time"
+
+	"github.com/lvl484/positioning-filter/config"
+	"github.com/lvl484/positioning-filter/storage"
 )
 
 func main() {
+	configPath := flag.String("cp", "../config", "Path to config file")
+	configName := flag.String("cn", "viper.config", "Name of config file")
+
+	flag.Parse()
+
+	viper, err := config.NewConfig(*configName, *configPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	consulConfig := viper.NewConsulConfig()
+	agentConfig := consulConfig.AgentConfig()
+	consulClient, err := consulConfig.NewClient()
+
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	if err = consulConfig.ServiceRegister(consulClient, agentConfig); err != nil {
+		log.Println(err)
+		return
+	}
+
+	defer consulClient.Agent().ServiceDeregister(consulConfig.ServiceName)
```

Comment:
Consul can deregister services automatically and timeout configurable

Reply:
added auto deregister in consul agent